EC-Switch-Seclog.py

```diff
@@ -0,0 +1,973 @@
+    # associate guardduty with target SECLOG
+    activate_guardduty(account_id, account_email,account_session,tseclog_id,tseclog_session)
+    # associate securityhub with target SECLOG
+    activate_securityhub(account_id, account_email,account_session,tseclog_id,tseclog_session)
+    
+    print("")
+    print(f"####### AWS Landing Zone switch SECLOG script finished. Executed in {time.time() - start_time} seconds")
+    print("#######")
+    print("")
+
+def usage():
+    """
+    This function prints the script usage
+    """
+    print('Usage:')
+    print('')
+    print('python EC-Switch-Seclog.py -a <account profile> [-s <seclog profile>] [-v]')
+    print('')
+    print('   Provide ')
+    print('   -a --account         : The AWS profile of the linked account to be switched')
+    print('   -s --sseclog         : The AWS profile of the source SECLOG account')
+    print('   -t --tseclog         : The AWS profile of the target SECLOG account')
+    print('   -o --org             : The AWS profile of the Organisation account')
+    print('   -v --verbose         : Debug mode - optional')
+
+
+def get_account_id(session):
+    """
+    This function gets te id of the account defined in the profile
+        :param force: flag to force the retrieval of the account ID 
+        :return: a string with the account id 
+    """
+
+    try:
+        sts = session.client('sts')
+        response = sts.get_caller_identity()
+        return response['Account']
+    except ClientError as error:
+        if error.response['Error']['Code'] == 'AccessDenied':
+            print(f"Access denied getting account id [{Status.FAIL.value}]")
+            print("Exiting...") 
+            sys.exit(1)
+        else:
+            raise error
+
+def is_seclog(session):
+    """
+    Function that checks if the account is a seclog account
+        :return: true or false
+    """
+    client = session.client('ssm')
+    seclog_account_id = get_account_id(session)
+    response = client.get_parameter(Name='/org/member/SecLogMasterAccountId')
+    if not 'Value' not in response or seclog_account_id != response['Parameter']['Value']:
+        return False
+    return True
+
+def remove_stacks_from_stackset(stackset,account_id, session):
+    
+    global regions
+
+    print(f"Remove stacks from StackSet {stackset} in progress ", end="")
+
+    cfn = session.client('cloudformation')
+    response = cfn.describe_stack_set(StackSetName=stackset)
+   
+    if response['StackSet']['Status'] not in ('ACTIVE'):
+        print(f"Cannot remove stacks from stackset {stackset}. Current stackset status is: {response['StackSet']['Status']} [{Status.FAIL.value}]")
+        print("Exiting...")
+        sys.exit(1)
+   
+    with Spinner():
+        filter=[{
+            'Name': 'DETAILED_STATUS',
+            'Values': 'PENDING'
+        }]
+        response = cfn.list_stack_instances(StackSetName=stackset,Filters=filter)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The API method `list_stack_instances` returns paginated results instead of all results. Consider using the pagination API or checking one of the following keys in the response to verify that all results were returned: `IsTruncated`, `NextToken`.

[Learn more](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/paginators.html)